how-to/workspace-platform-starter/client/src/framework/platform/platform-override.ts

```diff
@@ -864,6 +873,37 @@ export function overrideCallback(
 		public async saveDockProviderConfig(config: DockProviderConfigWithIdentity): Promise<void> {
 			return saveConfig(config, async (providerConfig) => super.saveDockProviderConfig(providerConfig));
 		}
+
+		/**
+		 * Determine whether or not a dialog should be shown.
+		 * @param payload the page that is going to be closed.
+		 * @returns Whether or not a modal should be shown
+		 */
+		public async handleSaveModalOnPageClose(
+			payload: HandleSaveModalOnPageClosePayload
+		): Promise<SaveModalOnPageCloseResult> {
+			// close confirmation modal will not be shown if the page is locked
+			if (isEmpty(unsavedPagePromptStrategy)) {
```

Comment:
Could this be done at the beginning of the overrideCallback instead of calling getSettings in here

Reply:
Done